korman/ui/modifiers/gui.py

```diff
@@ -67,3 +67,58 @@ def journalbookmod(modifier, layout, context):
 
     main_col.label("Clickable Region:")
     main_col.prop(modifier, "clickable_region", text="")
+
+def linkingbookmod(modifier, layout, context):
+    layout.prop_menu_enum(modifier, "versions")
+    layout.separator()
+
+    row = layout.row()
+    row.alert = modifier.clickable is None
+    row.prop(modifier, "clickable")
+    layout.prop(modifier, "clickable_region")
+
+    if "pvMoul" in modifier.versions:
+        row = layout.row()
+        row.alert = modifier.seek_point is None
+        row.prop(modifier, "seek_point")
+        layout.prop(modifier, "anim_type")
+
+        layout.separator()
+        layout.prop(modifier, "link_type")
+        row = layout.row()
+        row.alert = not modifier.age_instance
+        row.prop(modifier, "age_instance")
+        if modifier.link_type == "kChildAgeBook":
+            row = layout.row()
+            row.alert = not modifier.age_parent
+            row.prop(modifier, "age_parent")
+        if modifier.link_type == "kBasicLink":
+            row = layout.row()
+            row.alert = not modifier.age_uuid
+            row.prop(modifier, "age_uuid")
+
+    row = layout.row()
+    row.alert = not modifier.age_name
+    row.prop(modifier, "age_name")
```

Comment:
Looks like this same antipattern appears 6 times. Maybe break it out into a function?
```python
def linkingbookmod(...):
    def row_alert(prop_name, **kwargs):
        row = layout.row()
        row.alert = not getattr(modifier, prop_name)
        row.prop(modifier, prop_name, **kwargs)
    ...
```

Reply:
This was one of several repetitive patterns I was noticing.  I know I mentioned at least one in IRC.